Updated utils.common and rrt.py, rrt_test.py accordingly.

Srujan-D opened the pull request:
1) Updated the Node class in common.py
2) All the tuple objects have been replaced by Point() objects.
3) Some of the arguments like start_point, end_point and the elements of obstacle_list are expected to be in Point().
4) The sampler function as well as check_intersection function are expected to work with Point() and not tuple. (These functions as before are to be given by the user.)
5) The Node class is removed and can be imported from common.py available in utils. 
6) There is an option to use 3D coordinates, albeit some of the code will have to be updated in this case. Default z should be 0.

Srujan-D committed:
Updated common.py with Node class

Srujan-D committed:
Added restructured rrt.py

Srujan-D committed:
Delete rrt_restructured.py

Updated rrt.py directly.

Srujan-D committed:
Update rrt.py

1) All the tuple objects have been converted to Point().
2) Some of the arguments are expected to be in Point().
3) The sampler function as well as check_intersection function are expected to work with Point().
4) The Node class is removed and can be imported from common.py available in utils. 
5) There is an option to use 3D coordinates. Default z should be 0.  Some of the code will have to be updated in this case.

Srujan-D committed:
Update common.py

threewisemonkeys-as commented on gennav/planners/rrt/rrt.py:
```diff
@@ -1,43 +1,15 @@
-
-
-# TODO #32
 class RRT:
```
Could you inherit the `Planner` class from `planners/base.py` in this?

threewisemonkeys-as commented on gennav/planners/rrt/rrt.py:
```diff
@@ -60,15 +32,15 @@ def plan(self, start_point, goal_point, obstacle_list, animation=False):
         Args:
-            start_point: tuple with start point coordinates.
-            end_point: tuple with end point coordinates.
+            start_point: Point with start point coordinates.
```
Try to be specific here with `gennav.utils.geometry.Point` since even shapely has a point object. This applies to all docstrings

Srujan-D replied:
Yeah, makes sense. On it.

threewisemonkeys-as reviewed (commented):
Good work btw.

Also check out the lint and testing errors

Srujan-D committed:
Update rrt.py

1. Inherited Planner class from base.py
2. Updated the doc-strings

Srujan-D committed:
Rectified line causing import error.

Srujan-D committed:
Removed Environment() import.

threewisemonkeys-as commented on gennav/planners/rrt/rrt.py:
```diff
@@ -48,27 +18,25 @@ class RRT:
-
+            start_point: Point (from gennav.utils.geometry) with start point coordinates.
+            end_point: Point (from gennav.utils.geometry) with end point coordinates.
+            env: Base class for an envrionment.
```
Could you change the type for env to gennav.envs.Environment

threewisemonkeys-as commented:
Also I think in a lot of places you have used node youu did node.position which should be node.state.position

Srujan-D committed:
Update rrt.py

1) Corrected <Node_obj>.position to <Node_obj>.state.position
2) Added type of env (gennav.envs.Environment)
3) Imported Environment to use get_traj_status method properly.

Srujan-D committed:
Update rrt.py

1) Allowed user to add his own sampler.
2) Rectified lines regarding env.

Srujan-D committed:
Rectified Planner class Inheritance error

Srujan-D committed:
Update rrt_test.py

Updated rrt_test.py according to its restructured dependencies.

Srujan-D committed:
Update rrt_test.py

Converted path to Trajectory for it to pass successfully to path_optimizer method.

Srujan-D committed:
Formatted the code

Used black.

threewisemonkeys-as commented on gennav/planners/rrt/rrt.py:
```diff
@@ -1,46 +1,18 @@
-# from gennav.planners.samplers.samplers import uniform_adjustable_random_sampler as sampler
+# from gennav.utils.samplers import uniform_adjustable_random_sampler as sampler
 
+env = Environment()
```
this is unecessary

threewisemonkeys-as commented on gennav/planners/rrt/rrt.py:
```diff
@@ -1,46 +1,18 @@
-from gennav.utils.planner import check_intersection
-from matplotlib import pyplot as plt
-from shapely.geometry import Point, Polygon
+from gennav.envs import Environment
```
this is unecessary

threewisemonkeys-as commented on tests/test_planner/rrt_test.py:
```diff
@@ -14,16 +15,25 @@ def test_rrt():
         ],
     ]
 
+    poly = PolygonEnv(Environment)
```
You dont need to pass Environment into PolygonEnv

threewisemonkeys-as commented on tests/test_planner/rrt_test.py:
```diff
@@ -14,16 +15,25 @@ def test_rrt():
+            node = RobotState(position=node.state.position)
+        path = Trajectory(path)
+        
+        optimized_path = path_optimizer(path, poly)
```
No need to call this in rrt_test, it can be tested separately

Srujan-D replied:
Cool, I'll _remove_ the line then. It was there in the former rrt_test.

threewisemonkeys-as commented on gennav/planners/rrt/rrt.py:
```diff
@@ -152,50 +144,8 @@ def plan(self, start_point, goal_point, obstacle_list, animation=False):
-            RRT.visualize_tree(node_list, obstacle_list)
+        path.append(start)
 
         return list(reversed(path)), node_list
```
this method should only return a trajectory

threewisemonkeys-as commented on tests/test_planner/rrt_test.py:
```diff
@@ -14,16 +15,25 @@ def test_rrt():
         # Instatiate rrt planner object
         my_tree = RRT(sample_area=(-5, 15), sampler=sampler, expand_dis=0.1)
-        path, node_list = my_tree.plan((1, 1), (10, 10), obstacle_list)
+        path, node_list = my_tree.plan((1, 1), (10, 10), poly)
```
This will need to be modified since plan should only return a trajectory

Srujan-D replied:
Cool, I'll remove the `return node_list`.

threewisemonkeys-as commented on tests/test_planner/rrt_test.py:
```diff
@@ -14,16 +15,25 @@ def test_rrt():
+        # Convert path (list of Nodes) to Tajectory (list of RobotStates) 
+        for node in path:
+            node = RobotState(position=node.state.position)
+        path = Trajectory(path)
```
This will become uneccesary when plan returns a trajectory

Srujan-D committed:
Rectified lint errors.

Srujan-D committed:
Implemented the suggested changes.

1) Removed the import Environment.
2) Now the plan method only return a Trajectory- path

Srujan-D committed:
Implemented the suggested changes.

1) Removed the optimize_path and the corresponding lines.
2) Rectified poly = Polygon() line.

threewisemonkeys-as commented:
Btw you can click resolve n the changes here so I know what has been resolved

threewisemonkeys-as committed:
Merge branch 'restructure-base' into restruct_sruj

Srujan-D committed:
Cleared lint error.

Srujan-D committed:
Added PolygonEnv

Srujan-D committed:
Updated PolygonEnv import.
